app/org/sagebionetworks/bridge/services/TaskService.java

```diff
@@ -170,10 +168,9 @@ public void deleteTasks(String healthCode) {
     }
     
     private Activity createResponseActivityIfNeeded(StudyIdentifier studyIdentifier, String healthCode, Activity activity) {
-        if ((activity.getActivityType() != ActivityType.SURVEY)) {
+        if (activity.getActivityType() != ActivityType.SURVEY || activity.getSurveyResponse() != null) {
```

Comment:
Your statement from the documentation is correct. That line is intending to say "if this is a task activity, or you already have figured out the survey response for this activity, don't do anything". Let make it clearer.